met-web/src/components/engagement/form/EngagementForm.tsx

```diff
@@ -306,11 +306,12 @@ const EngagementForm = () => {
                         <Then>
                             <PrimaryButton
                                 sx={{ marginRight: 1 }}
-                                data-testid="engagement-form/create-engagement-button"
                                 onClick={() => handleCreateEngagement()}
                                 loading={isSaving}
                             >
-                                Create Engagement Draft
+                                <React.Fragment data-testid="engagement-form/create-engagement-button">
```

Comment:
What makes us have to create a React Fragment so we can pass data test id, can't we keep it in the primary button and get the button by it in the test and check for its existence?

Reply:
Ill remove that I was trying it out since data-testid was still giving me test errors on the Primary button

Reply:
@saravanpa-aot it was label changes originally but I started getting some jest errors so need to update the tests

Reply:
hm..i think i will get to the bottom of the issue if the test works now in current state and a small text change breaks the tests...

Reply:
if a text change requires tests to be modified , thats an improvement area for sure..We can discuss this

Reply:
ok sounds good